gui/default/syncthing/core/syncthingController.js

```diff
@@ -3124,6 +3124,132 @@ angular.module('syncthing.core')
                     address.indexOf('unix://') == 0 ||
                     address.indexOf('unixs://') == 0);
         };
+
+        $scope.shareDeviceIdDialog = function (method) {
+            // This function can be used to share both user's own and remote
+            // device IDs. Three sharing methods are used - copy to clipboard,
+            // send by email, and send by SMS.
+            var params = {
+                method: method,
+            };
+            var deviceID = $scope.currentDevice.deviceID;
+            var deviceName = $scope.deviceName($scope.currentDevice);
+
+            // Title and footer can be reused between different sharing
+            // methods, hence we define them separately before the body.
+            var title = $translate.instant('Syncthing device ID for "{%devicename%}"', {devicename: deviceName});
+            var footer = $translate.instant("Learn more at {%url%}.", {url: "https://syncthing.net"});
+
+            switch (method) {
+                case "email":
+                    params.heading = $translate.instant("Share by Email");
+                    params.icon = "fa fa-envelope-o";
+                    // Email message format requires using CRLF for line breaks.
+                    // Ref: https://datatracker.ietf.org/doc/html/rfc5322
+                    params.subject = title;
+                    params.body = [
+                        $translate.instant('"{%devicename%}" has shared their Syncthing device ID!', {devicename: deviceName}),
+                        $translate.instant("To connect with them, add a new device using the ID below."),
+                        deviceID,
+                        $translate.instant("Syncthing is a continuous file synchronization program. It synchronizes files between two or more computers in real time, safely protected from prying eyes. Your data is your data alone and you deserve to choose where it is stored, whether it is shared with some third party, and how it's transmitted over the internet."),
+                        footer
+                    ].join('\r\n\r\n');
+                    break;
+                case "sms":
+                    params.heading = $translate.instant("Share by SMS");
+                    params.icon = "fa fa-comments-o";
+                    // SMS is limited to 160 characters (non-Unicode), so we keep
+                    // it as short as possible, e.g. by stripping hyphens from
+                    // device ID. The current minimum length is around 140 chars,
+                    // but some room is required for longer sharing device names.
+                    params.body = [
+                        title + ":",
+                        deviceID.replace(/-/g, ''),
+                        footer
+                    ].join('\n\n');
```

Comment:
The message has only three-lines, we can spare the two extra newlines to gain some more headroom.

```suggestion
                    ].join('\n');
```

Reply:
Thanks.  In addition to saving the two characters, it looks much better now, at least in my SMS app view.

Reply:
Eh, I still think it's kind of cramped, especially on smaller screens, where the ID wraps along multiple lines, but I've checked SMS that I receive from banks and such, and they also put a lot of content without any empty lines there, so I guess that's the way things are normally done 🤔.